### PR TITLE
ARP: implement decoder and logger v2

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6827

Describe changes:
- Remove `hw_size` and `hw_size`
- Decode ARP pkts over GRE
 - Disable arp logger by default
 - Add `description` fields to schema.json

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1711

### CODE REVIEW

File: src/output-json-arp.c [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    }
+}
+
+static int JsonArpLogger(ThreadVars *tv, void *thread_data, const Packet *p)
+{
+    OutputJsonThreadCtx *thread = thread_data;
+    char srcip[46] = "";
+    char dstip[46] = "";
+
+    JsonBuilder *jb = CreateEveHeader((Packet *)p, LOG_DIR_PACKET, "arp", NULL, thread->ctx);
```
Comment (victorjulien) [inline src/output-json-arp.c:221]: this cast is wrong, `CreateEveHeader` takes a const Packet pointer. `git grep` suggest we have more offending code.

File: src/output-json-arp.c [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+{
+    switch (proto) {
+        case 0x0800:
+            return "ipv4";
+        default:
+            return "unknown";
+    }
+}
+
+static const char *opcode2str(uint16_t opcode)
```
Comment (victorjulien) [inline src/output-json-arp.c:154]: these functions don't match our naming style

File: src/output-json.c [hunk 3]
```diff
@@ -714,8 +714,7 @@ void CreateEveFlowId(JsonBuilder *js, const Flow *f)
     }
 }
 
-void JSONFormatAndAddMACAddr(JsonBuilder *js, const char *key,
-                             uint8_t *val, bool is_array)
+void JSONFormatAndAddMACAddr(JsonBuilder *js, const char *key, uint8_t *val, bool is_array)
```
Comment (victorjulien) [inline src/output-json.c:717]: unrelated change to this commit?

File: doc/userguide/upgrade.rst [hunk 4]
```diff
@@ -50,6 +50,8 @@ Major changes
 - ``SIP_PORTS`` variable has been introduced in suricata.yaml
 - Application layer's ``sip`` counter has been split into ``sip_tcp`` and ``sip_udp``
   for the ``stats`` event.
+- Decoder and logger for ARP protocol has been introduced.
```
Comment (victorjulien) [inline doc/userguide/upgrade.rst:53]: can you also document the logger in the yaml docs and the eve record in the eve format docs?

**Review by victorjulien — CHANGES_REQUESTED** [review]
Nice work Giuseppe. Some comments/questions inline.

File: src/output-json.c (hunk 3 above)
Comment (glongo) [reply src/output-json.c:717]: Yes sorry.

File: doc/userguide/upgrade.rst (hunk 4 above)
Comment (glongo) [reply doc/userguide/upgrade.rst:53]: Right, sure.
